doc/adrs/107_add_support_for_transient_data_in_forms.adoc

```diff
@@ -0,0 +1,84 @@
+While the data in the various widgets would come from the model, the currently selected object should not be persisted in the model at all.
+
+
+== Decision
+
+We will add the ability for specifiers to create forms allowing the manipulation of transient data not backup by a model.
+
+Let's consider a very simple text field allowing the manipulation of a transient piece of text.
+
+In order to allow such use case, we will have to specify how to retrieve the value of the textfield.
+For that, we will keep everything we have with the variable manager and the value provider (for the core API) and the value expression (for the view DSL).
+We will only have to provide some additional variables to retrieve the transient data and we will be able to display some text in our text field.
+
+We will thus have to figure out how to define, initialize and update those variables.
+Since we want those variables to be transient, they will be specific to every single instance of a representation.
+For example, two users watching the same form with a master / details user interface could have selected a different value.
+
+=== Transient datastore and lifecycle of the representation
+
+We could store those transient data in the frontend.
+This woule easily ensure that they are specific to the instance of the representation.
+They would be provided to the backend as subscription parameters using the input in a similar fashion as the list of expanded tree items in a tree-based representation.
+
+Such an approach would have a couple of downsides.
+Firstly, we would be limited to basic types of transient data (string, boolean, number, list) and it would thus not be possible to have transient model objects or custom types.
+Specifiers would only have the ability to manipulate a set of unrelated primitive value.
+If you think about the new element dialog, we could not have a transient model element (with its behavior such as derived fields etc) but instead a transient piece of data for each widget of such form.
+
+Secondly, we will have to send to the backend all those data, everytime they change.
+This will imply sending potentially a sizable amount of data in the subscription.
+We may also spend our time subscribing / unsubscribing to various state of the representation on the server just to propagate a change in a value.
+
+While we are currently doing that for tree based representations in order to help share instances of the representation between users who are viewing the same state (same list of expanded tree items), as of today, we cannot conclude in any way that it is better than having one instance of the representation per subscription.
+End users can quickly end up with different list of expanded tree items so we may end up with one instance of the representation per subscription in practice anyway.
+On top of that, the cost of subscribing / unsubscribing may be more expensive than what we believe.
```

Comment:
And we know it's not always as reliable as we want (at least using Apollo Client).